test/unit/advertisement-operator/controller_test.go

```diff
@@ -0,0 +1,126 @@
+package advertisement_operator
+
+import (
+	v1 "github.com/liqoTech/liqo/api/advertisement-operator/v1"
+	policyv1 "github.com/liqoTech/liqo/api/cluster-config/v1"
+	advcontroller "github.com/liqoTech/liqo/internal/advertisement-operator"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"testing"
+)
+
+func createReconciler(acceptedAdv, maxAcceptableAdv int32, autoAccept bool) advcontroller.AdvertisementReconciler {
+	return advcontroller.AdvertisementReconciler{
+		Client:           nil,
+		Log:              ctrl.Log.WithName("controller-test"),
+		Scheme:           nil,
+		EventsRecorder:   nil,
+		KubeletNamespace: "",
+		KindEnvironment:  false,
+		VKImage:          "",
+		InitVKImage:      "",
+		HomeClusterId:    "",
+		AcceptedAdvNum:   acceptedAdv,
+		ClusterConfig: policyv1.ClusterConfigSpec{
+			MaxAcceptableAdvertisement: maxAcceptableAdv,
+			AutoAccept:                 autoAccept,
+		},
+	}
+}
+
+func createAdvertisement() v1.Advertisement {
+	return v1.Advertisement{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{},
+		Spec:       v1.AdvertisementSpec{},
+		Status:     v1.AdvertisementStatus{},
+	}
+}
+
+func TestCheckAdvertisementWithAutoAccept(t *testing.T) {
+	r := createReconciler(0, 10, true)
+
+	// ACCEPT TEST
+	// given a configuration with max 10 Advertisements, create 10 Advertisements and check that they are all accepted
+	for i := 0; i < 10; i++ {
+		adv := createAdvertisement()
+		r.CheckAdvertisement(&adv)
+		assert.Equal(t, "ACCEPTED", adv.Status.AdvertisementStatus)
+	}
+	// check that the Adv counter has been incremented
+	assert.Equal(t, int32(10), r.AcceptedAdvNum)
+
```

Comment:
I would suggest to have one single testCase per Test in order to: (1) foster parallelism and (2) early detection of which testCase is really broken. In this case, one for the ShouldAccept one and one for ShouldRefuse